raiden-cli/src/routes/userDeposit.ts

```diff
@@ -0,0 +1,151 @@
+  const totalDepositTarget = parseAsBigNumber(totalDeposit);
+  const currentTotalDeposit = await this.raiden.getUDCCapacity();
+  const depositDifference = totalDepositTarget.sub(currentTotalDeposit);
+
+  if (depositDifference.lte(Zero)) throw TotalDepositTooLowError;
+
+  const transactionHash = await this.raiden.depositToUDC(depositDifference);
+  return { transaction_hash: transactionHash };
+}
+
+async function planWithdraw(this: Cli, amount: string): Promise<PlanWithdrawResponse> {
+  const transactionHash = await this.raiden.planUDCWithdraw(amount);
+  const withdrawPlan = await this.raiden.getUDCWithdrawPlan();
+  return {
+    transaction_hash: transactionHash,
+    planned_withdraw_block_number: withdrawPlan!.block, // The plan must exists here.
+  };
+}
+
+async function withdraw(this: Cli, amount: string): Promise<WithdrawResponse> {
+  const transactionHash = await this.raiden.withdrawFromUDC(amount);
+  return { transaction_hash: transactionHash };
+}
+
+async function determineAndExecuteRequestedInteraction(
+  this: Cli,
+  request: Request,
+): Promise<DepositResponse | PlanWithdrawResponse | WithdrawResponse> {
+  if (shouldDeposit(request)) {
+    const totalDeposit = getTotalDepositAmount(request)!; // Can't be undefined here.
+    return await deposit.call(this, totalDeposit);
+  } else if (shouldPlanWithdraw(request)) {
+    const amount = getPlannedWithdrawAmount(request)!; // Can't be undefined here.
+    return await planWithdraw.call(this, amount);
+  } else if (shouldWithdraw(request)) {
+    const amount = getWithdrawAmount(request)!; // Can't be undefined here.
+    return await withdraw.call(this, amount);
+  } else {
+    throw MalformedRequestFormat;
+  }
+}
+
+async function getUDCWithdrawPlan(
+  this: Cli,
+  _request: Request,
+  response: Response,
+): Promise<void> {
+  const withdrawPlan = await this.raiden.getUDCWithdrawPlan();
+  response.json(withdrawPlan);
+}
+
+async function interactWithUDC(this: Cli, request: Request, response: Response): Promise<void> {
+  try {
+    const interactionResponse = await determineAndExecuteRequestedInteraction.call(this, request);
+    response.json(interactionResponse);
+  } catch (error) {
+    if (
+      isInvalidParameterError(error) ||
+      error === MalformedNumberValue ||
+      error === MalformedRequestFormat
+    ) {
+      response.status(400).send(error.message);
+    } else if (isInsuficientFundsError(error)) {
+      response.status(402).send(error.message);
+    } else if (isConflictError(error) || error === TotalDepositTooLowError) {
+      response.status(409).send(error.message);
+    } else {
+      response.status(500).send(error.message);
+    }
+  }
+}
+
+/**
+ * @param this - Cli object
+ * @returns Router instance
+ */
+export function makeUserDepositRouter(this: Cli): Router {
+  const router = Router();
+
+  router.get('/withdrawPlan', getUDCWithdrawPlan.bind(this));
```

Comment:
Is this endpoint also provided like that by the PC API? If not, I think we could/should keep consistency on the naming and use snake_case for the endpoint paths

Reply:
✔️